charms/jupyter-controller/contributing.md

```diff
@@ -0,0 +1,75 @@
+# Contributing
+
+## Overview
+
+This document outlines the processes and practices recommended for contributing enhancements to jupyter_controller.
+
+## Talk to us First
+
+Before developing enhancements to this charm, you should [open an issue](/../../issues) explaining your use case. If you would like to chat with us about your use-cases or proposed implementation, you can reach us at [MLOps Mattermost public channel](https://chat.charmhub.io/charmhub/channels/mlops-documentation) or on [Discourse](https://discourse.charmhub.io/).
+
+## Pull Requests
+
+Please help us out in ensuring easy to review branches by rebasing your pull request branch onto the `main` branch. This also avoids merge commits and creates a linear Git commit history.
+
+All pull requests require review before being merged. Code review typically examines:
+  - code quality
+  - test coverage
+  - user experience for Juju administrators of this charm.
+
+## Recommended Knowledge
+
+Familiarising yourself with the [Charmed Operator Framework](https://juju.is/docs/sdk) library will help you a lot when working on new features or bug fixes.
+
+## Build Charm
+
+To build jupyter_controller run:
+
+```shell
+charmcraft pack
+```
+
+## Developing
+
+You can use the environments created by `tox` for development. For example, to load the `lint` environment into your shell, run:
+
+```shell
+tox --notest -e lint
+source .tox/lint/bin/activate
+```
+
+### Testing
+
+Use tox for testing. For example to test test, run:
```

Comment:
I really don't like the way this reads. We could obv change the testenv name to something else, but this probably is an issue with the template, I mean we could have something like `{{ tox_example_test_env }}` there. WDYT?